web/degausser.js

```diff
@@ -2,11 +2,16 @@ globalThis.degausser = input => {
   var div = document.createElement("div");
   div.innerHTML = input;
   document.body.appendChild(div);
-  Array.from(div.querySelectorAll("a, img, area, input")).forEach(el => {
+  Array.from(div.querySelectorAll("a, input")).forEach(el => {
     if (el.childNodes.length === 0) {
       el.outerHTML = el.alt || "";
     }
   });
+  Array.from(div.querySelectorAll("img, area, image")).forEach(el => {
+    if (el.childNodes.length === 0) {
```

Comment:
What is the child node check for?

Reply:
I guess it has the same reason as the code on line https://github.com/flowpub/degausser/pull/101/files#diff-fc4356b3cdc9231a897b27c239fd5647af047b6aea5bd6bc11319521cda49604R6
I would suggest to use https://developer.mozilla.org/ru/docs/Web/API/HTMLElement/innerText
to check if either an image element or its children have a text representation that is visible on the page

Reply:
I separated the `"img, area, image"` items from the forEach loop above to account for the changes to alt text. I'd rather not change this since it's not relevant to this PR.

Reply:
I mean on line 11 it's better to check whether the image has a text representation or not. Because checking for children can't answer this question I assume? e.g.  `<img ....><p></p></img>` - the image have neither alt text, nor text inside its children, but `el.childNodes.length` this will return 1, because of <p>?

Reply:
Hmm actually now that I'm looking at it does this even apply to these elements? I don't think `img`, `area`, and `image` elements can eevn have child elements. I think I can remove the if condition completely.

Reply:
Yeah. It seems that according to the specs the `img` tag shouldn't have content. Hence, the condition should be omitted
https://developer.mozilla.org/en-US/docs/Web/CSS/Replaced_element
https://www.littlewebhut.com/css/info_nonreplaced_replaced/
https://catalin.red/html-replaced-void-elements/#:~:text=According%20to%20W3C%2C%20an%20HTML,the%20HTML%20void%20elements%20too.